src/ComponentProvider.ts

```diff
@@ -1,94 +1,60 @@
-import type { AsyncAPIDocument, ChannelParameter, Message, MessageTrait, Schema } from '@asyncapi/parser';
-
+import type {
+  AsyncAPIDocument,
+  ChannelParameter,
+  Message,
+  Schema,
+} from '@asyncapi/parser';
+
+import { JSONPath } from 'jsonpath-plus';
+import _ from 'lodash';
 /**
  * This class will provide all sorts of data for optimizers.
  *
  * @private
  */
 export class ComponentProvider {
+  messagePaths = ['$.channels.*.*.message', '$.components.messages.*'];
```

Comment:
Remember that `operation.[publish | subscribe]` can have message defined as list, as `oneOf` - https://github.com/asyncapi/spec/blob/master/spec/asyncapi.md#operationObject

Reply:
I am doing it in a hacky way. open for suggestions...